tutorials/flex_ro/unit_models.py

```diff
@@ -0,0 +1,344 @@
+    ei_lb, ei_ub = params.get_energy_intensity_bounds()
+    for skid in blk.set_ro_skids:
+        blk.ro_skid[skid].recovery.setlb(params.minimum_recovery)
+        blk.ro_skid[skid].recovery.setub(params.maximum_recovery)
+        blk.ro_skid[skid].energy_intensity.setlb(ei_lb)
+        blk.ro_skid[skid].energy_intensity.setub(ei_ub)
+
+
+def posttreatment_operation_model(blk, params: um_params.FlexDesalParams):
+    """
+    Builds operation model for the posttreatment unit
+
+    Parameters
+    ----------
+    blk : OperationModel
+        IDAES OperationModel instance
+
+    params : object
+        Input parameters needed for the model
+    """
+    _add_required_variables(blk)
+
+    blk.energy_intensity.fix(params.posttreatment.energy_intensity)
+    blk.del_component(blk.recovery)
+    blk.del_component(blk.calculate_product_flowrate)
+
+    # If the posttreatment unit is not operating, then set product flowrate to zero
+    # Connect post-treatment operation with the RO startup variables
+    blk.suppress_product_flowrate = Constraint(
+        expr=blk.product_flowrate <= params.intake.nominal_flowrate * blk.op_mode
+    )
+    blk.suppress_reject_flowrate = Constraint(
+        expr=blk.reject_flowrate <= params.intake.nominal_flowrate * (1 - blk.op_mode)
+    )
+
+    # Update the power consumption calculation constraint
+    blk.calculate_power_consumption.set_value(
+        blk.power_consumption == blk.energy_intensity * blk.feed_flowrate,
+    )
+
+
+def brine_discharge_operation_model(blk, params: um_params.FlexDesalParams):
+    """
+    Builds operation model for the brine discharge unit
+
+    Parameters
+    ----------
+    blk : OperationModel
+        IDAES OperationModel instance
+
+    params : object
+        Input parameters needed for the model
+    """
+    # Declare model parameters
+    blk.energy_intensity = Param(
+        initialize=params.brinedischarge.energy_intensity,
+        units=pyunits.kWh / pyunits.m**3,
+        mutable=True,
+    )
+
+    # Declare essential variables
+    blk.feed_flowrate = Var(within=NonNegativeReals, units=pyunits.m**3 / pyunits.hr)
+    blk.power_consumption = Var(within=NonNegativeReals, units=pyunits.kW)
+
+    # the brine sump only consumes power if the RO is off,
+    # otherwise brine is pushed out by the leftover RO pressure
+    blk.calculate_power_consumption = Constraint(
+        expr=blk.power_consumption
+        >= blk.energy_intensity
+        * (blk.feed_flowrate + params.intake.nominal_flowrate * (blk.op_mode - 1)),
+        doc="Power requirement for brine discharge",
+    )
+
+
+def power_generation_operation_model(blk, params: um_params.FlexDesalParams):
+    """
+    Builds the operation model for onsite power generation
+
+    Parameters
+    ----------
+    blk : OperationModel
+        Instance of IDAES OperationModel
+
+    design_blk : DesignModel
+        Design model containing information on the peak capacity
```

Comment:
I didn't notice the use of DesignModel anywhere in these files. Are we only using OperationModels? I assume this should be deleted.